lib/broadcast_manager.dart

```diff
@@ -34,12 +43,9 @@ class BroadcastManager {
     if (!_pendingBroadcast) {
       _pendingBroadcast = true;
 
-      // Schedule a broadcast event to be run on the microtask queue. The broadcast is run
-      // async so that multiple broadcast events can be batched together into one update
-      // across all changes that occur in the current task of the event loop.
-      scheduleMicrotask(() {
-        _broadcast();
-      });
+      // The broadcast is run async so that multiple broadcast events can be batched
+      // together into one update across all changes that occur in the current task of the event loop.
+      Future.delayed(Duration.zero, _broadcast);
```

Comment:
Doing this in a micro task was blocking the main thread. Scheduling it on the event queue instead of the micro task queue is the more reasonable priority level for this.